### PR TITLE
Some fixes related to help command

### LINKED ISSUE #366: Include 'help', if we have other commands at the top level
If other top-level commands are present, the displayed list of Available Commands should include the `help` command—however, it doesn't.

Since this repository seems the correct place for this issue, I mention it here (see Hugo issue [2349](https://github.com/spf13/hugo/issues/2349)).

### LINKED ISSUE #393: Misbehaviour of IsHelpCommand()
I notice, what IsHelpCommand always returns false even on built-in help command and I found out why:
Look at the docs of IsHelpCommand:
```
IsHelpCommand determines if a command is a 'help' command; a help command is
determined by the fact that it is NOT runnable/hidden/deprecated, and has no
sub commands that are runnable/hidden/deprecated.
```
In the docs it says, what help is not runnable, but if you look in `initHelpCmd` definition
```go

func (c *Command) initHelpCmd() {
	if c.helpCommand == nil {
		if !c.HasSubCommands() {
			return
		}

		c.helpCommand = &Command{
			Use:   "help [command]",
			Short: "Help about any command",
			Long: `Help provides help for any command in the application.
    Simply type ` + c.Name() + ` help [path to command] for full details.`,
			PersistentPreRun:  func(cmd *Command, args []string) {},
			PersistentPostRun: func(cmd *Command, args []string) {},

			Run: func(c *Command, args []string) {
				cmd, _, e := c.Root().Find(args)
				if cmd == nil || e != nil {
					c.Printf("Unknown help topic %#q.", args)
					c.Root().Usage()
				} else {
					cmd.Help()
				}
			},
		}
	}
	c.AddCommand(c.helpCommand)
}
```
, you can see, what built-in help command has Run function.
So the actual logic of IsHelpCommand is wrong. I think, the possible solution could be:
```go
func (c *Command) IsHelpCommand() bool {
    helpCmd := c.findHelpCommand()
    if c == helpCmd{
        return true
    }
    return false
}

func (c Command) findHelpCommand() *Command {
    if c.helpCommand == nil {
        if c.HasParent() {
             return c.Parent().findHelpCommand()
        }
    }
    return c.helpCommand
}
```

### PR DESCRIPTION
**Pull request by bogem**
Fixes #366

### CODE REVIEW

**Comment by eparis** [comment]
lets talk in the #393 issue...  I don't think this is right, but I agree it is terribly confusing.

**Comment by bogem** [comment]
Changed PR so it fixes only #366.
Fix of #393 is discussed ibid

File: command.go [hunk 1]
```diff
@@ -783,7 +782,8 @@ func (c *Command) initHelpCmd() {
 			Run: func(c *Command, args []string) {
 				cmd, _, e := c.Root().Find(args)
 				if cmd == nil || e != nil {
-					c.Printf("Unknown help topic %#q.", args)
+					c.Printf("Unknown help topic %#q", args)
```
Comment (eparis) [inline command.go]: Can't we just use `"Unknown help topic %%q\n"`?

Comment (bogem) [inline command.go]: Yes, fixed

File: command.go [hunk 2]
```diff
@@ -783,7 +782,7 @@ func (c *Command) initHelpCmd() {
 			Run: func(c *Command, args []string) {
 				cmd, _, e := c.Root().Find(args)
 				if cmd == nil || e != nil {
-					c.Printf("Unknown help topic %#q.", args)
+					c.Printf("Unknown help topic %%q\n", args)
```
Comment (eparis) [inline command.go]: you just went from `%#q` to `%%q` which doesn't seem to be right.

Comment (bogem) [inline command.go]: Now everything seems work

File: command.go [hunk 3]
```diff
@@ -783,7 +782,7 @@ func (c *Command) initHelpCmd() {
 			Run: func(c *Command, args []string) {
 				cmd, _, e := c.Root().Find(args)
 				if cmd == nil || e != nil {
-					c.Printf("Unknown help topic %#q.", args)
+					c.Printf("Unknown help topic %q\n", args)
```
Comment (eparis) [inline command.go]: Why the change from `%#q` to   `%q` ? Not that I have a strong opion about `"` quoting vs `` ` `` quoting. But it seems like a needless change...

Comment (bogem) [inline command.go]: But you said in the first review about it.

Comment (bogem) [inline command.go]: 😅

Comment (eparis) [inline command.go]: Blech, I made a mistype in the first review. I wanted you to use `\n` and I mistyped `%%q\n`. Which now I realize is why you used `%%`.

How about `%#q\n`?  That work?

Comment (bogem) [inline command.go]: Yes, it works. 
How it looks like in my CLI:

**Pull request merged**